src/monero_fmt_plug.c

```diff
@@ -0,0 +1,270 @@
+/*
+ * Format for cracking Monero wallet hashes.
+ *
+ * This software is Copyright (c) 2017, Dhiru Kholia <dhiru [at] openwall.com>,
+ * and it is hereby released to the general public under the following terms:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ */
+
+#if FMT_EXTERNS_H
+extern struct fmt_main fmt_monero;
+#elif FMT_REGISTERS_H
+john_register_one(&fmt_monero);
+#else
+
+#include <string.h>
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+#define OMP_SCALE               1  // MKPC and OMP_SCALE tuned on i5-6500 CPU
+
+#include "formats.h"
+#include "misc.h"
+#include "common.h"
+#include "params.h"
+#include "options.h"
+#include "chacha.h"
+#include "memdbg.h"
+
+#define FORMAT_LABEL            "monero"
+#define FORMAT_NAME             "monero Wallet"
+#define FORMAT_TAG              "$monero$"
+#define TAG_LENGTH              (sizeof(FORMAT_TAG) - 1)
+#define ALGORITHM_NAME          "AES+ChaCha+Various 64/" ARCH_BITS_STR
```

Comment:
We don't use to have `+` in algo names. I'd prefer spaces or perhaps `/`

Reply:
...and perhaps it should say "pseudo AES" or something, since it's not a normal one.